src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,109 @@
+import { FC, useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from "../../api";
+
+export const PeoplePage: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const { personSlug } = useParams();
+
+  const personInfo = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const personName = (name: string | null) => {
+    return peoples.find(person => person.name === name);
```

Comment:
You are using the `find` method which will iterate over the entire array until it finds the first match. If the array is large, this could be inefficient. Consider using a different data structure, like a Map or an Object, where the keys are the names of the people. This would allow you to access a person by name in constant time.